crates/ruff_python_parser/src/lib.rs

```diff
@@ -417,6 +417,15 @@ impl Tokens {
         };
         &self[start..=start + end]
     }
+
+    /// Returns a slice of tokens after the given [`TextSize`] offset.
+    ///
+    /// If the given offset lies within a token, the returned slice will start from the token after
+    /// that. In other words, the returned slice will not include the token containing the offset.
+    pub fn after(&self, offset: TextSize) -> &[Token] {
```

Comment:
Do we depend on this behavior? I think I would rather have the implementation `panic` if the offset isn't at a token boundary.

Reply:
I think I'll have a clearer picture once the code compiles. I plan on being able to do that by today as that will allow me to do a lot of testing. And, I plan on revisiting the APIs once everything is changed.